### PR TITLE
Add support for basic music controls

### PR DESCRIPTION
**Pull request by postlund**
It is now possible to use basic music controls like play(), pause(), stop(),
next() and previous() for the currently active source.

NB: Will only do something useful if the active source supports it.

### CODE REVIEW

**Comment by wuub** [comment]
I'll take a look tomorrow :) :+1: Right now it LGTM, so it should land in  v0.2.1

**Comment by postlund** [comment]
Great, hope it's good enough :) Looking forward to a new release so that I can improve support in Home-Assistant 👍

**Comment by wuub** [comment]
FYI @sdague is also working on improving HA yamaha support (especially regarding multi zone support added in https://github.com/wuub/rxv/pull/22 / v0.2.0 release). If you have not seen tha,t you might want to take a look to not redo any work already done :)

File: rxv/rxv.py [hunk 1]
```diff
@@ -124,6 +125,30 @@ def on(self, state):
...
+    def stop(self):
+        self._send_action('Stop')
+
+    def next(self):
+        self._send_action('Skip Fwd')
+
+    def previous(self):
+        self._send_action('Skip Rev')
+
+    def _send_action(self, action):
```
Comment (wuub) [inline rxv/rxv.py]: Rename it to something more specific. `_playback_control` or sth :) `_send_action` is too  generic

**Comment by postlund** [comment]
Yeah, I noticed that and most of the changes do not intersect so we should complement each other quite nicely :)

File: rxv/rxv.py [hunk 2]
```diff
@@ -124,6 +125,30 @@ def on(self, state):
...
+        src_name = self._src_name(self.input)
+        if not src_name:
+            return None
```
Comment (wuub) [inline rxv/rxv.py]: I'm not sure, but I'm feeling we should probably raise some kind of exception here `PlaybackUnavailable` or sth when sth goes wrong.

Also having a function to check if playback actions are available would also make API simpler to use.

File: rxv/rxv.py (hunk 1 above)
Comment (postlund) [inline rxv/rxv.py]: Sure thing, that's better :)

File: rxv/rxv.py (hunk 2 above)
Comment (postlund) [inline rxv/rxv.py]: I was thinking about the same thing but afaik the only way to know if playback control is available is to check the input source. So we would have to compare the current input source with a list of pre-defined sources that we know support this. Not pretty solution but I guess that could work as long as we can find a complete list.

Comment (wuub) [inline rxv/rxv.py]: :+1:

**Comment by wuub** [comment]
https://pypi.python.org/pypi/rxv/0.3.0 :)

**Comment by postlund** [comment]
Cool! :) Thanks for the fast responses!

**Comment by sdague** [comment]
I'm pretty sure the capabilities for this can be discovered on the desc.xml. I know NET RADIO is weird, and stop works, but then play doesn't, at least on my receiver. And next / prev aren't supported. 

This should be good enough to get started, but if anyone wants to enhance it further, I'd look at the desc.xml data.

**Pull request merged**